modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -1214,6 +1217,13 @@ public void start(
                 if (isFeatureEnabled(IGNITE_DISTRIBUTED_META_STORAGE_FEATURE))
                     ((DistributedMetaStorageImpl)ctx.distributedMetastorage()).inMemoryReadyForRead();
 
+                mntcProcessor.registerWorkflowCallbackIfTaskExists(ClearFolderWorkflow.CLEAR_FOLDER_TASK, task -> {
```

Comment:
I don't think that `IgniteKernal` is the right place to register a workflow. Perhaps, `CacheProcessor.start()` would be the best choice.